cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -336,54 +338,49 @@ std::vector<std::string> aggregate_reader_metadata::get_pandas_index_names() con
   return names;
 }
 
-std::tuple<size_type, size_type, std::vector<row_group_info>>
+std::tuple<int64_t, size_type, std::vector<row_group_info>>
 aggregate_reader_metadata::select_row_groups(
   host_span<std::vector<size_type> const> row_group_indices,
-  size_type row_start,
-  size_type row_count) const
+  int64_t skip_rows_opt,
+  std::optional<size_type> const& num_rows_opt) const
 {
   std::vector<row_group_info> selection;
+  auto [rows_to_skip, rows_to_read] = [&]() {
+    if (not row_group_indices.empty()) { return std::pair<int64_t, size_type>{}; }
+    auto const from_opts = cudf::io::detail::skip_rows_num_rows_from_options(
+      skip_rows_opt, num_rows_opt, get_num_rows());
+    return std::pair{static_cast<int64_t>(from_opts.first), from_opts.second};
```

Comment:
It looks like we can't return directly here solely because we need to cast the uint to int, is that right?